src/libs/PersonalDetailsUtils.ts

```diff
@@ -24,15 +24,27 @@ Onyx.connect({
     },
 });
 
+// Index for the substring method to remove the merged account prefix.
+const substringStartIndex = CONST.MERGED_ACCOUNT_PREFIX.length;
+
 function getDisplayNameOrDefault(passedPersonalDetails?: Partial<PersonalDetails> | null, defaultValue = '', shouldFallbackToHidden = true, shouldAddCurrentUserPostfix = false): string {
-    let displayName = passedPersonalDetails?.displayName ? passedPersonalDetails.displayName.replace(CONST.REGEX.MERGED_ACCOUNT_PREFIX, '') : '';
+    let displayName = passedPersonalDetails?.displayName ?? '';
+
+    // If the displayName starts with the merged account prefix, remove it.
+    if (displayName.startsWith(CONST.MERGED_ACCOUNT_PREFIX)) {
+        displayName = displayName.substring(substringStartIndex);
+    }
+
     if (shouldAddCurrentUserPostfix && !!displayName) {
         displayName = `${displayName} (${Localize.translateLocal('common.you').toLowerCase()})`;
     }
 
-    const fallbackValue = shouldFallbackToHidden ? Localize.translateLocal('common.hidden') : '';
-
-    return displayName || defaultValue || fallbackValue;
+    // If displayName exists, return it early so we don't have to allocate
```

Comment:
NAB but I don't think this comment makes much sense without the context of the code that was here before, so I think it would be fine to remove it.